tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/errorconstructorexpr/ErrorConstructorExprTest.java

```diff
@@ -99,8 +99,13 @@ public void testErrorConstructorExprNegative() {
         validateError(negativeSemanticResult, i++, "cannot infer type of the error from '(error|ErrorB)'", 51, 20);
         validateError(negativeSemanticResult, i++, "unknown type 'Blah'", 60, 27);
         validateError(negativeSemanticResult, i++, "cannot create a new error value from 'ErrorU1'", 62, 19);
+        validateError(negativeSemanticResult, i++, "incompatible types: expected 'error', found 'ErrorU2'", 63, 13);
```

Comment:
This isn't correct, right? The RHS expression itself is erroneous, so we can't mention RHS is `ErrorU2`. IMO just the next error (cannot create a new error value from 'ErrorU2') is enough here. WDYT?

We can fix this separately too.

Reply:
Actually, one error is issued from here https://github.com/ballerina-platform/ballerina-lang/blob/8adfc0ac5b12d60c70d33893a15e49e7eba2920e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L2799.

And we return from that point. So currently when we provide `typeRef` there will be duplicates.